src/any_llm/providers/deepseek/utils.py

```
@@ -44,6 +44,29 @@ def _convert_structured_type_to_deepseek_json(
+    here from the ``extra_content["deepseek"]`` side-channel populated by
+    ``_inject_reasoning_extra_content`` when the response was first received.
+
+    Reference: https://api-docs.deepseek.com/guides/thinking_mode#tool-calls
+    """
+    result = []
+    for message in messages:
+        if message.get("role") == "assistant" and message.get("tool_calls") is not None:
+            extra_content = message.get("extra_content")
+            deepseek_extra = extra_content.get("deepseek") if isinstance(extra_content, dict) else None
+            if isinstance(deepseek_extra, dict) and isinstance(deepseek_extra.get("reasoning_content"), str):
+                message = {**message, "reasoning_content": deepseek_extra["reasoning_content"]}
```

Ruff (0.15.21)

[warning] `for` loop variable `message` overwritten by assignment target

(PLW2901)